h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstRBind.java

```diff
@@ -235,7 +235,6 @@ public void map(Chunk cs) {
       int idx = _chunkOffset + cs.cidx();
       Key ckey = Vec.chunkKey(_v._key, idx);
       if (_cmap != null) {
-        assert !cs.hasFloat() : "Input chunk (" + cs.getClass() + ") has float, but is expected to be categorical";
```

Comment:
This will fix the problem.

However I think there is a bigger issue here. The Categorical chunk filled with NAs has no idea it holds categorical values and will return true for hasFloats() which doesn't make sense.

This can IMHO happen only for categorical chunks with NAs only. Instead of removing the assert completely we can also check if it is full of NAs.